### PR TITLE
check for undefined values on setPluginResolvedOn function

### LINKED ISSUE #7959: [Plugin] Cannot set property of undefined
### Current behavior:
On cypress startup, it throws "Cannot set property [key] of undefined" error when modifying config from plugin.
This happens when setting a new object property to config on plugins. 

### Desired behavior:
Cypress startup properly and loads config on plugins.

### Test code to reproduce
add code to **/plugins/index.js** file and start cypress

```js
module.exports = (on, config) => {
  config.test = { foo: 'bar' }; // Fails
  config.test2 = "foobar"; // Pass
  config.test3 = { foo: {bar: "baz"} }; // Fails
  config.test4 = ["foo", "bar", "baz"]; // Pass
  return config;
};
```

### Versions
Cypress: 4.10.0 
OS: Manjaro/Arc linux

┆Issue is synchronized with this [Jira Bug](https://cypress-io.atlassian.net/browse/TR-115) by [Unito](https://www.unito.io/learn-more)

### PR DESCRIPTION
**Pull request by jmsansan**
- Closes #7959 

### User facing changelog

Fixed a bug where recursion function setPluginResolvedOn could receive undefined on resolvedObj parameter.

### CODE REVIEW

**Comment by jmsansan** [comment]
> Thanks for the contribution! Could you please sign our CLA?
> 
> Additionally, we will need some tests written to verify this new behavior.
> 
> There is a Contributing Guide that covers how to contribute and get Cypress running locally here: https://github.com/cypress-io/cypress/blob/develop/.github/CONTRIBUTING.md
> 
> Let us know if you have trouble getting the tests running locally.

I will do my best since is my first contribution to an open source project :+1: 
Thanks Jennifer.

**Comment by jennifer-shehane** [comment]
@jmsansan Thanks for signing the CLA, unfortunately the author of all of the commits does not match the email address on your GitHub account. There are 2 commits, one made by Jose Sanchez the other made by jmsansan. https://github.com/cypress-io/cypress/pull/7960/commits

Our CLA bot gets confused when they don't match. Here are some solutions to fixing when commits are linked to another wrong user: https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/

**Comment by jmsansan** [comment]
@jennifer-shehane 
I could figure out what was the problem with CLA so I solved it, PR should be fine now.
Thanks for your support Jennifer :+1:

File: packages/server/test/unit/config_spec.js [hunk 1]
```diff
@@ -1309,6 +1309,31 @@ describe('lib/config', () => {
         },
       })
     })
+
+    it('resolves a single object', () => {
```
Comment (jennifer-shehane) [inline packages/server/test/unit/config_spec.js:1313]: We like to link to the original issues when adding a test so we have reference later why we needed this. 

```suggestion
    // https://github.com/cypress-io/cypress/issues/7959
    it('resolves a single object', () => {
```

**Comment by mccataldo** [comment]
@jennifer-shehane should this also resolve #7002 and #1736

**Comment by jennifer-shehane** [comment]
@mccataldo Yes! It does appear to have also fixed those issues. Nice catch 👍 I'll update those other issues to reflect that they've been fixed.

**Pull request merged**